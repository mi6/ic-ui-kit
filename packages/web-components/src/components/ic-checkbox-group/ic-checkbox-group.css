@import "../../global/normalize.css";

:host {
  display: block;
}

ic-input-label.error {
  color: var(--ic-status-error);
}

ic-input-label ic-typography {
  margin-bottom: var(--ic-space-sm);
}

:host(.small) ic-input-label ic-typography {
  margin-bottom: calc(var(--ic-space-sm) / 2);
}

ic-input-validation {
  margin-top: var(--ic-space-sm);
}

:host(.small) ic-input-validation {
  margin-top: calc(var(--ic-space-sm) / 2);
}

.checkboxes-container {
  margin-bottom: calc(-1 * var(--ic-space-xxs));
}

:host(.small) .checkboxes-container {
  margin-bottom: calc(-1 * var(--ic-space-xxxs));
}

.screen-reader-only-text {
  position: absolute;
  left: -9999px;
  background-color: #fff;
  color: #000;
  text-transform: none;
}
