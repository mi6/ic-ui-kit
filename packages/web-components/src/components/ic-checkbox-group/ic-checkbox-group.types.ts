export interface IcChangeEventDetail {
  value: string[];
  checkedOptions: {
    checkbox: HTMLIcCheckboxElement;
    textFieldValue?: string;
  }[];
  selectedOption: HTMLIcCheckboxElement;
}
