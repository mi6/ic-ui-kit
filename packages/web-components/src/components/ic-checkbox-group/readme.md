# ic-checkbox-group



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute           | Description                                                                                                                         | Type                                      | Default     |
| -------------------- | ------------------- | ----------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------- | ----------- |
| `disabled`           | `disabled`          | If `true`, the checkbox group will be set to the disabled state.                                                                    | `boolean`                                 | `false`     |
| `helperText`         | `helper-text`       | The helper text that will be displayed for additional field guidance.                                                               | `string`                                  | `""`        |
| `hideLabel`          | `hide-label`        | If `true`, the label will be hidden and the required label value will be applied as an aria-label.                                  | `boolean`                                 | `false`     |
| `label` _(required)_ | `label`             | The label for the checkbox group to be displayed.                                                                                   | `string`                                  | `undefined` |
| `name` _(required)_  | `name`              | The name for the checkbox group to differentiate from other groups.                                                                 | `string`                                  | `undefined` |
| `required`           | `required`          | If `true`, the checkbox group will require a value.                                                                                 | `boolean`                                 | `false`     |
| `size`               | `size`              | The size of the checkboxes to be displayed. This does not affect the font size of the label.                                        | `"default" \| "large" \| "small"`         | `"default"` |
| `small`              | `small`             | <span style="color:red">**[DEPRECATED]**</span> This prop should not be used anymore. Set prop `size` to "small" instead.<br/><br/> | `boolean`                                 | `false`     |
| `validationStatus`   | `validation-status` | The validation status - e.g. 'error' \| 'warning' \| 'success'.                                                                     | `"" \| "error" \| "success" \| "warning"` | `""`        |
| `validationText`     | `validation-text`   | The validation text - e.g. 'error' \| 'warning' \| 'success'.                                                                       | `string`                                  | `""`        |


## Events

| Event      | Description                         | Type                               |
| ---------- | ----------------------------------- | ---------------------------------- |
| `icChange` | Emitted when a checkbox is checked. | `CustomEvent<IcChangeEventDetail>` |


## Dependencies

### Depends on

- [ic-input-label](../ic-input-label)
- [ic-input-validation](../ic-input-validation)

### Graph
```mermaid
graph TD;
  ic-checkbox-group --> ic-input-label
  ic-checkbox-group --> ic-input-validation
  ic-input-label --> ic-typography
  ic-input-validation --> ic-typography
  style ic-checkbox-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


