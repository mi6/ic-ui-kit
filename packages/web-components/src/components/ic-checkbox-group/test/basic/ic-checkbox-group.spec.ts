import { newSpecPage } from "@stencil/core/testing";
import { CheckboxGroup } from "../../ic-checkbox-group";
import { Checkbox } from "../../../ic-checkbox/ic-checkbox";
import { TextField } from "../../../ic-text-field/ic-text-field";

describe("ic-checkbox-group", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test">
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders");
  });

  it("should render as required", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test" required>
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-required");
  });

  it("should render as helper text", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test" helper-text="helper test">
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-helpertext");
  });

  it("should render checked", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test">
        <ic-checkbox value="test" checked label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-checked");
  });

  it("should render indeterminate", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test">
        <ic-checkbox value="test" checked indeterminate label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-indeterminate");
  });

  it("should render an aria label when label hidden", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" hide-label name="test">
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-hidden-label");
  });

  it("should render in a disabled state", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test">
        <ic-checkbox value="test" disabled label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-disabled");
  });

  it("should render with validation status", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" hide-label name="test" validation-status="error">
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-validation");
  });

  it("should render with hidden dynamic additional field", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox, TextField],
      html: `<ic-checkbox-group label="test label" hide-label name="test" validation-status="error">
        <ic-checkbox value="test" additional-field-display="dynamic" label="test label">
        <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-dynamic-field");
  });

  it("should render with dynamic additional field when checked", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox, TextField],
      html: `<ic-checkbox-group label="test label" hide-label name="test" validation-status="error">
        <ic-checkbox value="test" label="test label" additional-field-display="dynamic" checked>
        <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot(
      "renders-with-dynamic-field-when-checked"
    );
  });

  it("should render with disabled static additional field", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox, TextField],
      html: `<ic-checkbox-group label="test label" hide-label name="test" validation-status="error">
        <ic-checkbox value="test" label="test label" additional-field-display="static">
        <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-disabled-additional-field");
  });

  it("should remove disabled attribute from static additional field when checked", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox, TextField],
      html: `<ic-checkbox-group label="test label" hide-label name="test" validation-status="error">
        <ic-checkbox value="test" label="test label" additional-field-display="static" checked>
        <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-checkbox>    
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot(
      "renders-with-disabled-attribute-removed-from-field"
    );
  });

  it("should pass the size on the checkbox group to the child checkboxes when there's no size set on them individually", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test" size="small">
        <ic-checkbox value="test" label="test label"></ic-checkbox>
        <ic-checkbox value="test" label="test label"></ic-checkbox>  
      </ic-checkbox-group>`,
    });

    expect(page.root).toMatchSnapshot();

    const checkboxes = page.root.querySelectorAll("ic-checkbox");

    checkboxes.forEach((checkbox) => {
      expect(checkbox.classList.contains("small")).toBe(true);
    });
  });

  it("should prioritise the size on an individual checkbox over the size on the checkbox group", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test" size="large">
        <ic-checkbox value="test" label="test label" id="large-checkbox"></ic-checkbox>
        <ic-checkbox value="test" label="test label" size="small" id="small-checkbox"></ic-checkbox>  
      </ic-checkbox-group>`,
    });

    const smallCheckbox = page.root.querySelector(
      "ic-checkbox#small-checkbox"
    ) as HTMLIcCheckboxElement;
    expect(smallCheckbox["size"]).toMatch("small");

    const largeCheckbox = page.root.querySelector(
      "ic-checkbox:not(#small-checkbox)"
    ) as HTMLIcCheckboxElement;
    expect(largeCheckbox.classList.contains("large")).toBe(true);

    expect(page.root).toMatchSnapshot();
  });

  it("should change checked state", async () => {
    const page = await newSpecPage({
      components: [Checkbox],
      html: `<ic-checkbox value="test" label="test label" checked></ic-checkbox>`,
    });

    expect(page.root.checked).toBe(true);

    page.rootInstance.handleClick();
    await page.waitForChanges();

    expect(page.root.checked).toBe(false);
  });

  it("should call 'setFocus' when checkbox is focused", async () => {
    const page = await newSpecPage({
      components: [Checkbox],
      html: `<ic-checkbox value="test" label="test label" checked></ic-checkbox>`,
    });

    //Can't expect anything in this test - this is to increase code coverage only
    await page.rootInstance.setFocus().toHaveBeenCalled;
  });

  it("should emit icCheck on click", async () => {
    const page = await newSpecPage({
      components: [CheckboxGroup, Checkbox],
      html: `<ic-checkbox-group label="test label" name="test">
        <ic-checkbox value="test" label="test label"></ic-checkbox>    
      </ic-checkbox-group>`,
    });

    const eventSpy = jest.fn();

    page.win.addEventListener("icCheck", eventSpy);

    page.root
      .querySelector("ic-checkbox")
      .shadowRoot.querySelector("input")
      .click();

    await page.waitForChanges();

    expect(eventSpy).toHaveBeenCalled();
  });

  it("should test form reset event", async () => {
    const page = await newSpecPage({
      components: [Checkbox],
      html: `<form>
          <ic-checkbox value="test" label="test label" checked></ic-checkbox>    
          <button id="resetButton" type="reset">Reset</button>  
      </form>`,
    });

    expect(page.root.checked).toBe(true);

    page.root.checked = false;
    await page.waitForChanges();

    expect(page.root.checked).toBe(false);

    await page.rootInstance.handleFormReset();
    await page.waitForChanges();

    expect(page.root.checked).toBe(true);

    //test disconnected callback
    page.setContent("");
  });
});
