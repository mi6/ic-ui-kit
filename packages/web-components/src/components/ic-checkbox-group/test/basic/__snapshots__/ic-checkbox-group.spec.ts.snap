// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-checkbox-group should pass the size on the checkbox group to the child checkboxes when there's no size set on them individually 1`] = `
<ic-checkbox-group class="small" label="test label" name="test" size="small">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="small" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
  <ic-checkbox additional-field-display="static" class="small" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should prioritise the size on an individual checkbox over the size on the checkbox group 1`] = `
<ic-checkbox-group class="large" label="test label" name="test" size="large">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="large" id="large-checkbox" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
  <ic-checkbox additional-field-display="static" class="small" id="small-checkbox" label="test label" size="small" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should remove disabled attribute from static additional field when checked: renders-with-disabled-attribute-removed-from-field 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test" validation-status="error">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label invalid data
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText test-validation-text" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
    <ic-input-validation arialivemode="polite" for="test" message="" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" checked="" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <svg class="checkmark" clip-rule="evenodd" fill-rule="evenodd" height="1.5rem" viewBox="0 0 24 24" width="1.5rem" xmlns="http://www.w3.org/2000/svg">
          <title>
            checkmark icon
          </title>
          <path d="M21 6.285l-11.16 12.733-6.84-6.018 1.319-1.49 5.341 4.686 9.865-11.196 1.475 1.285z"></path>
        </svg>
        <input checked="" class="checkbox checked" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container">
        <div>
          <div class="additional-field-wrapper">
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container>
          <ic-input-label for="ic-text-field-input-3" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-3" inputmode="text" name="ic-text-field-input-3" placeholder="Placeholder" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
      <input class="ic-input" name="ic-text-field-input-3" type="hidden" value="">
    </ic-text-field>
    <input class="ic-input" name="test" type="hidden" value="test">
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render an aria label when label hidden: renders-hidden-label 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render as helper text: renders-with-helpertext 1`] = `
<ic-checkbox-group class="default" helper-text="helper test" label="test label" name="test">
  <mock:shadow-root>
    <fieldset aria-labelledby="test-helper-text" id="test">
      <legend>
        <ic-input-label for="test" helpertext="helper test" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render as required: renders-required 1`] = `
<ic-checkbox-group class="default" label="test label" name="test" required="">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label required
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label" required=""></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render checked: renders-checked 1`] = `
<ic-checkbox-group class="default" label="test label" name="test">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" checked="" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <svg class="checkmark" clip-rule="evenodd" fill-rule="evenodd" height="1.5rem" viewBox="0 0 24 24" width="1.5rem" xmlns="http://www.w3.org/2000/svg">
          <title>
            checkmark icon
          </title>
          <path d="M21 6.285l-11.16 12.733-6.84-6.018 1.319-1.49 5.341 4.686 9.865-11.196 1.475 1.285z"></path>
        </svg>
        <input checked="" class="checkbox checked" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
    <input class="ic-input" name="test" type="hidden" value="test">
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render in a disabled state: renders-disabled 1`] = `
<ic-checkbox-group class="default" label="test label" name="test">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default disabled" disabled="" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" disabled="" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render indeterminate: renders-indeterminate 1`] = `
<ic-checkbox-group class="default" label="test label" name="test">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" checked="" class="default" indeterminate="" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <div class="indeterminate-symbol"></div>
        <input checked="" class="checkbox checked indeterminate" id="ic-checkbox-test-label-test-label" indeterminate="" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
    <input class="ic-input" name="test" type="hidden" value="test">
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render with disabled static additional field: renders-with-disabled-additional-field 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test" validation-status="error">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label invalid data
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText test-validation-text" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
    <ic-input-validation arialivemode="polite" for="test" message="" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container">
        <div>
          <div class="additional-field-wrapper">
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field disabled="" label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container disabled="">
          <ic-input-label disabled="" for="ic-text-field-input-2" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container disabled="" validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" disabled="" id="ic-text-field-input-2" inputmode="text" name="ic-text-field-input-2" placeholder="" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
      <input class="ic-input" disabled="" name="ic-text-field-input-2" type="hidden" value="">
    </ic-text-field>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render with dynamic additional field when checked: renders-with-dynamic-field-when-checked 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test" validation-status="error">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label invalid data
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText test-validation-text" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
    <ic-input-validation arialivemode="polite" for="test" message="" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="dynamic" checked="" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <svg class="checkmark" clip-rule="evenodd" fill-rule="evenodd" height="1.5rem" viewBox="0 0 24 24" width="1.5rem" xmlns="http://www.w3.org/2000/svg">
          <title>
            checkmark icon
          </title>
          <path d="M21 6.285l-11.16 12.733-6.84-6.018 1.319-1.49 5.341 4.686 9.865-11.196 1.475 1.285z"></path>
        </svg>
        <input checked="" class="checkbox checked" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container" style="display: flex;">
        <div class="branch-corner"></div>
        <div>
          <ic-typography variant="caption">
            <p aria-live="polite" class="dynamic-text">
              This selection requires additional answers
            </p>
          </ic-typography>
          <div>
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container>
          <ic-input-label for="ic-text-field-input-1" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-1" inputmode="text" name="ic-text-field-input-1" placeholder="Placeholder" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
      <input class="ic-input" name="ic-text-field-input-1" type="hidden" value="">
    </ic-text-field>
    <input class="ic-input" name="test" type="hidden" value="test">
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render with hidden dynamic additional field: renders-with-dynamic-field 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test" validation-status="error">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label invalid data
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText test-validation-text" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
    <ic-input-validation arialivemode="polite" for="test" message="" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="dynamic" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container" style="display: none;">
        <div class="branch-corner"></div>
        <div>
          <ic-typography variant="caption">
            <p aria-live="polite" class="dynamic-text">
              This selection requires additional answers
            </p>
          </ic-typography>
          <div>
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container>
          <ic-input-label for="ic-text-field-input-0" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-0" inputmode="text" name="ic-text-field-input-0" placeholder="Placeholder" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
      <input class="ic-input" name="ic-text-field-input-0" type="hidden" value="">
    </ic-text-field>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render with validation status: renders-with-validation 1`] = `
<ic-checkbox-group class="default" hide-label="" label="test label" name="test" validation-status="error">
  <mock:shadow-root>
    <span aria-hidden="true" class="screen-reader-only-text" id="screenReaderOnlyText">
      test label invalid data
    </span>
    <fieldset aria-labelledby="screenReaderOnlyText test-validation-text" id="test">
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
    <ic-input-validation arialivemode="polite" for="test" message="" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;

exports[`ic-checkbox-group should render: renders 1`] = `
<ic-checkbox-group class="default" label="test label" name="test">
  <mock:shadow-root>
    <fieldset aria-labelledby="" id="test">
      <legend>
        <ic-input-label for="test" helpertext="" label="test label"></ic-input-label>
      </legend>
      <div class="checkboxes-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ic-checkbox additional-field-display="static" class="default" label="test label" value="test">
    <mock:shadow-root>
      <div class="container">
        <input class="checkbox" id="ic-checkbox-test-label-test-label" name="test" role="checkbox" type="checkbox" value="test">
        <ic-typography class="checkbox-label" variant="body">
          <label htmlfor="ic-checkbox-test-label-test-label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-checkbox>
</ic-checkbox-group>
`;
